frontend/src/lib/components/InsightLegend/InsightLegendButton.tsx

```diff
@@ -1,18 +1,65 @@
 import { Button } from 'antd'
 import { useActions, useValues } from 'kea'
 import { IconLegend } from 'lib/lemon-ui/icons'
+import { insightDataLogic } from 'scenes/insights/insightDataLogic'
 import { insightLogic } from 'scenes/insights/insightLogic'
-import { isFilterWithDisplay } from 'scenes/insights/sharedUtils'
+import { TrendsFilterType } from '~/types'
 import { shouldShowLegend } from './InsightLegend'
 
+export function InsightLegendButtonDataExploration(): JSX.Element | null {
+    const { insightProps } = useValues(insightLogic)
+    const { insightFilter, hasLegend } = useValues(insightDataLogic(insightProps))
+    const { updateInsightFilter } = useActions(insightDataLogic(insightProps))
+
+    const showLegend = (insightFilter as TrendsFilterType)?.show_legend
+    const toggleShowLegend = (): void => {
+        updateInsightFilter({ show_legend: !showLegend })
+    }
+
+    return (
+        <InsightLegendButtonComponent
+            hasLegend={hasLegend}
+            showLegend={showLegend}
+            toggleShowLegend={toggleShowLegend}
+        />
+    )
+}
+
 export function InsightLegendButton(): JSX.Element | null {
     const { filters } = useValues(insightLogic)
     const { toggleInsightLegend } = useActions(insightLogic)
 
-    return shouldShowLegend(filters) && isFilterWithDisplay(filters) ? (
```

Comment:
`shouldShowLegend` already checks for `isFilterWithDisplay`, thus it could be removed here.